src/main/Extensions/EmptyTrash/EmptyTrash.ts

```diff
@@ -0,0 +1,53 @@
+import type { ExecOptions } from "child_process";
+import { exec } from "child_process";
+import { platform } from "os";
+
+interface CommandOptions {
+    command: string;
+    options?: ExecOptions;
+}
+
+export class EmptyTrash {
+    public static async emptyTrash(): Promise<boolean> {
+        return new Promise((resolve) => {
+            const command = this.getEmptyTrashCommand();
+
+            if (!command) {
+                resolve(false);
+                return;
+            }
+
+            exec(command.command, command.options || {}, (error: Error | null) => {
+                if (error) {
+                    console.error("Error emptying trash:", error);
+                    resolve(false);
+                    return;
+                }
+                resolve(true);
+            });
+        });
+    }
+
+    private static getEmptyTrashCommand(): CommandOptions | null {
+        const os = platform();
```

Comment:
Instead of `platform()` you can use the `OperatingSystem` type from `@common/Core`. Additionaly I encourage you to use the `CommandlineUtility` from `@Core/CommandlineUtility` and the `PowershellUtility` from `@Core/PowershellUtility` instead of using `exec`. Here is an example:

EmptyTrash.ts:

```typescript
import type { OperatingSystem } from "@common/Core";
import type { CommandlineUtility } from "@Core/CommandlineUtility";
import type { PowershellUtility } from "@Core/PowershellUtility";

export class EmptyTrash {
    public constructor(
        private readonly commandlineUtility: CommandlineUtility,
        private readonly powershellUtility: PowershellUtility,
        private readonly operatingSystem: OperatingSystem,
    ) {}

    public async emptyTrash() {
        const map: Record<OperatingSystem, () => Promise<void>> = {
            Linux: async () => {
                await this.commandlineUtility.executeCommand("rm -rf ~/.local/share/Trash/*");
            },
            macOS: async () => {
                await this.commandlineUtility.executeCommand(
                    "osascript -e 'tell application \"Finder\" to activate' -e 'tell application \"Finder\" to if ((count of items in trash) > 0) then empty trash'",
                );
            },
            Windows: async () => {
                await this.powershellUtility.executeCommand("Clear-RecycleBin -Force");
            },
        };

        await map[this.operatingSystem]();
    }
}
```

And here is the action handler:

EmptyTrashActionHandler.ts:

```typescript
import type { ActionHandler } from "@Core/ActionHandler";
import type { SearchResultItemAction } from "@common/Core";
import type { EmptyTrash } from "./EmptyTrash";

export class EmptyTrashActionHandler implements ActionHandler {
    public readonly id = "EmptyTrash";

    public constructor(private readonly emptyTrash: EmptyTrash) {}

    public async invokeAction(action: SearchResultItemAction): Promise<void> {
        if (action.argument !== "empty") {
            throw new Error(`Argument "${action.argument}" is not supported`);
        }

        await this.emptyTrash.emptyTrash();
    }
}
```

Here is how you can bootstrap the extension and the action handler:

EmptyTrashModule.ts:

```typescript
import type { Dependencies } from "@Core/Dependencies";
import type { DependencyRegistry } from "@Core/DependencyRegistry";
import type { ExtensionBootstrapResult } from "../ExtensionBootstrapResult";
import type { ExtensionModule } from "../ExtensionModule";
import { EmptyTrash } from "./EmptyTrash";
import { EmptyTrashActionHandler } from "./EmptyTrashActionHandler";
import { EmptyTrashExtension } from "./EmptyTrashExtension";

export class EmptyTrashModule implements ExtensionModule {
    public bootstrap(dependencyRegistry: DependencyRegistry<Dependencies>): ExtensionBootstrapResult {
        return {
            extension: new EmptyTrashExtension(dependencyRegistry.get("AssetPathResolver")),
            actionHandlers: [
                new EmptyTrashActionHandler(
                    new EmptyTrash(
                        dependencyRegistry.get("CommandlineUtility"),
                        dependencyRegistry.get("PowershellUtility"),
                        dependencyRegistry.get("OperatingSystem"),
                    ),
                ),
            ],
        };
    }
}
```